pre_commit_hooks/requirements_txt_fixer.py

```diff
@@ -21,8 +21,9 @@ def __init__(self):  # type: () -> None
     @property
     def name(self):  # type: () -> bytes
         assert self.value is not None, self.value
-        if self.value.startswith(b'-e '):
-            return self.value.lower().partition(b'=')[-1]
+        for egg in (b'#egg=', b'&egg='):
+            if egg in self.value:
+                return self.value.lower().partition(egg)[-1]
```

Comment:
it seems that `egg=` is optional when non-editable:

```
git+https://github.com/asottile/astpretty
```

```console
$ pip install -r t.txt
...
Successfully installed astpretty-1.6.0
```

Reply:
yes, egg is optional when installing in non-editable mode.
But in that case we don't really have a way determine the package name.